src/Avalonia.Base/Rendering/Composition/Server/ServerCustomCompositionVisual.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Numerics;
+using Avalonia.Logging;
+using Avalonia.Media;
+using Avalonia.Rendering.Composition.Transport;
+
+namespace Avalonia.Rendering.Composition.Server;
+
+internal class ServerCompositionCustomVisual : ServerCompositionContainerVisual, IServerClockItem
+{
+    private readonly CompositionCustomVisualHandler _handler;
+    private bool _wantsNextAnimationFrameAfterTick;
+    internal ServerCompositionCustomVisual(ServerCompositor compositor, CompositionCustomVisualHandler handler) : base(compositor)
+    {
+        _handler = handler;
+        _handler.Attach(this);
+    }
+
+    protected override void DeserializeChangesCore(BatchStreamReader reader, TimeSpan commitedAt)
+    {
+        base.DeserializeChangesCore(reader, commitedAt);
+        var count = reader.Read<int>();
+        for (var c = 0; c < count; c++)
+        {
+            try
+            {
+                _handler.OnMessage(reader.ReadObject()!);
+            }
+            catch (Exception e)
+            {
+                Logger.TryGet(LogEventLevel.Error, LogArea.Visual)
+                    ?.Log(_handler, $"Exception in {nameof(CompositionCustomVisualHandler)} {{0}}", e);
```

Comment:
I think we can log here type name of the handler. Would help a lot with diagnosing errors here.